api/all/src/main/java/io/opentelemetry/api/baggage/propagation/W3CBaggagePropagator.java

```diff
@@ -48,20 +50,35 @@ private W3CBaggagePropagator() {}
     if (baggage.isEmpty()) {
       return;
     }
+    String headerContent = baggageToString(baggage);
+
+    if (!headerContent.isEmpty()) {
```

Comment:
@breedx-splk Fixed some nits in https://github.com/open-telemetry/opentelemetry-java/pull/3871/commits/001f5aa241d1ab376b25cad46e359f02bb1d2d04